File: backend/schemas/news.py
from datetime import datetime
from typing import Optional, Dict
from pydantic import HttpUrl, BaseModel
from schemas.base import BaseSchema, BaseResponse

class NewsBase(BaseSchema):
    company_id: int
    headline: str
    summary: Optional[str] = None
    source: Optional[str] = None
    url: Optional[str] = None # Using string instead of HttpUrl to avoid strict validation errors on weird domains
    published_at: Optional[datetime] = None
    provider: Optional[str] = None
    retrieved_at: Optional[datetime] = None
    relevance_score: Optional[float] = None

class NewsResponse(NewsBase, BaseResponse):
    pass

class NewsStatisticsResponse(BaseSchema):
    total_articles: int
    articles_today: int
    articles_this_week: int
    companies_with_news: int
    companies_without_news: int
    average_articles_per_company: float
    average_relevance_score: float
    provider_distribution: Dict[str, int]
    average_refresh_duration_ms: float
    duplicate_percentage: float
    rejected_article_percentage: float
    
class NewsHealthResponse(BaseSchema):
    provider_status: str
    scheduler_status: str
    last_refresh_time: Optional[datetime]
    last_failure: Optional[str]
    pending_jobs: int

File: frontend/src/types/news.ts
// Fields match backend schemas/news.py NewsResponse
export interface NewsItem {
  id: number;
  company_id: number;
  headline: string;       // backend field name (not 'title')
  summary: string | null; // backend field name (not 'snippet' or 'ai_summary')
  source: string | null;  // backend field name (not 'source_provider')
  url: string | null;
  published_at: string | null;
  provider: string | null;
  retrieved_at: string | null;
  relevance_score: number | null;
  created_at: string;
  updated_at: string;
}

export interface NewsRunLog {
  id: number;
  started_at: string;
  completed_at: string | null;
  articles_fetched: number;
  articles_stored: number;
  provider: string | null;
  company_name: string | null;
  status: string;
  error_message: string | null;
}

export interface NewsStatisticsResponse {
  total_articles: number;
  articles_last_24h: number;
  articles_by_provider: Record<string, number>;
  articles_by_company: Record<string, number>;
  recent_runs: NewsRunLog[];
}

export interface NewsHealthResponse {
  provider_status: string;
  scheduler_status: string;
  last_refresh_time: string | null;
  last_failure: string | null;
  pending_jobs: number;
}
